src/client/requirements.js

```diff
@@ -0,0 +1,61 @@
+import { SECOND } from '../utils/constants';
+
+export const DEFAULTS = {
+	freshness: Number.MAX_SAFE_INTEGER,
+	timeout: 20 * SECOND,
+};
+
+/**
+ * Combines component requirements into a requirements list by endpoint.
+ * @param {Map} requirementsByComponent Key: Component, Value: requirements with endpoint/params property.
```

Comment:
Do we need to be concerned with doing any sanitization of inputs, i.e. validating what is passed in is a `Map` here?

Reply:
These functions are designed to be used by the higher order component and not so much directly. I'm not sure it's good to add KBs to the bundle for validation that shouldn't be needed.